packages/core-data/src/entities.js

```diff
@@ -218,6 +218,18 @@ export const rootEntitiesConfig = [
 	},
 ];
 
+export const deprecatedEntities = {
+	postType: {
+		attachment: {
+			alternative: {
```

Comment:
Do we really need another level and the `alternative` key? I'd expect we could just pass the recommended entity directly to `attachment`, like this:

```js
export const deprecatedEntities = {
	postType: {
		attachment: {
			kind: 'root',
			name: 'media',
			version: '6.9',
		},
	},
};
```

Reply:
That looks like a better option - I've updated the object to that shape

Reply:
Also just noticed that I need to change `version` to `since` in the config for deprecated entities as they mean different things (version is when it will be removed).

Rather than only supporting particular properties in the config, I've made it so that all the properties for the `deprecated` function are supported in case anyone wants to add `version`, `hint` and others in the future. `alternative` is the only one treated specially.